frodo.js

```diff
@@ -3,35 +3,34 @@
  *
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.